cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java

```diff
@@ -1169,18 +1186,38 @@ static boolean pullDependencyPackages(String orgName, String packageName, String
         if (packageCompilation.getResolution().diagnosticResult().hasErrors()) {
             return true;
         }
-
-        JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(packageCompilation, JvmTarget.JAVA_17);
-        Collection<Diagnostic> backendDiagnostics = jBallerinaBackend.diagnosticResult().diagnostics(false);
-        if (!backendDiagnostics.isEmpty()) {
-            printDiagnostics(backendDiagnostics);
+        if (!hasProvidedPlatformDeps(packageCompilation)) {
+            JBallerinaBackend jBallerinaBackend = JBallerinaBackend.from(packageCompilation, JvmTarget.JAVA_17);
+            Collection<Diagnostic> backendDiagnostics = jBallerinaBackend.diagnosticResult().diagnostics(false);
+            if (!backendDiagnostics.isEmpty()) {
+                printDiagnostics(backendDiagnostics);
+            }
+            return jBallerinaBackend.diagnosticResult().hasErrors();
         }
-        return jBallerinaBackend.diagnosticResult().hasErrors();
+        errStream.println("Warning: Cache generation skipped due to platform dependencies with 'provided' scope");
```

Comment:
```suggestion
        errStream.println("Warning: Cache generation skipped for platform dependencies with 'provided' scope");
```

Reply:
The cache generation is skipped for the pulled packages because platform dependencies are found in them and not for the platform dependencies. So does this change properly give that meaning?

Reply:
yeah it doesn't. Let's keep the message as is.